mne_bids/read.py

```diff
@@ -327,23 +327,24 @@ def _handle_info_reading(sidecar_fname, raw, verbose=None):
                         'the MEG JSON sidecar file corresponds to the raw '
                         'data for KIT files.')
         else:
-            hpi_freqs_json = sidecar_json['HeadCoilFrequency']
-            try:
-                hpi_freqs_raw, _, _ = mne.chpi.get_chpi_info(raw.info)
-            except ValueError:
-                logger.info(
-                    'Cannot verify that the cHPI frequencies provided in '
-                    'the MEG JSON sidecar file correspond to those in the '
-                    'raw data. (Was it converted from another format?)'
-                )
-            else:
-                if not np.allclose(hpi_freqs_json, hpi_freqs_raw):
-                    raise ValueError(
-                        f'The cHPI coil frequencies in the sidecar file '
-                        f'{sidecar_fname}:\n    {hpi_freqs_json}\ndiffer from'
-                        f' what is stored in the raw data:\n'
-                        f'    {hpi_freqs_raw}\nCannot proceed.'
+            if 'HeadCoilFrequency' in hpi_freqs_json:
```

Comment:
The check here is fine in principle, but the dict `hpi_freqs_json` does not exist, look at the line you are replacing:

`hpi_freqs_json = sidecar_json['HeadCoilFrequency']`

here you take the "HeadCoilFrequency" KEY from the sidecar_json DICT --> the value of HeadCoilFrequency happens to be a DICT itself, so previously `hpi_freqs_json` was a DICT.

But you have removed that line, so the only dict you have is sidecar_json.

Thus, your line should read: `if 'HeadCoilFrequency' in sidecar_json`

Reply:
Yeah, totally! I'm sorry, that was a stupid mistake. I'm p new to git so I would appreciate some help :). 

When I ran `pytest` it only gave me this warning:
`DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working`
So I figured my code was fine and that the warning came from some other part of the code that was already there before. 
Before I push the corrected version of my code, can maybe someone give me a few pointers on how to work with pytest? Because I followed the instruction in CONTRIBUTING.md (Running `pytest` from the command line while in the mne-bids dir) and like I mentioned above, it only gave me the DeprecationWarning. How can I get it to run all the tests, see the results as they are displayed under "Results" without having to push my code first? I would like to see if it works before creating a PR.

Reply:
Hey @fravona2211, I just merged PR #857, which updated the contributor documentation to be current (the previous one was slightly outdated).

Would you mind going through https://github.com/mne-tools/mne-bids/blob/main/CONTRIBUTING.md once more and following all the steps?

- Are you on linux, osx, or windows for your operating system?
- Do you use `conda` from the command line, or how are you managing your Python installations?

I can help you to get your development environment up and running if you run into problems with following the CONTRIBUTING.md file. Just let me know.

Reply:
PS: I recommend that you start from scratch (using `conda`)

Reply:
hi @sappelhoff :) 
I just deleted everything and then went throuh CONTRIBUTING.md once more and set everything up according to it (incl. MAKE). 
I am on Windows and used `conda` from the command line. 

Now that I seem to have everything set up according to the document, how I can use the test suite locally? Running pytest still returns the same result as in my last post (`DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working`). 

Thanks for your help!

Reply:
you should be able to run the test suite by:

1. opening the command line interface
2. activating the mne-bids conda environment (usually `conda activate <environment name>`, where you have your own "environment name"
3. then using `cd` to navigate to the root of the cloned mne-bids repository
4. finally calling `make test`

If you followed all other steps in CONTRIBUTING.md, that should work 🤔 

As for the result you get, I don't know what's wrong.

1. is the branch in your clone of `mne-bids` up to date?
2. what does `python --version` tell you (AFTER activating your conda environment)?

Reply:
Hi! 

I will start off with answering your questions:

1. I cloned it only yesterday, so I assume my branch is up to date. Otherwise, how could I check if it actually is? 
2. My python version in my conda environment is Python 3.9.6

When I call `make test`, it first sucessfully processes the dependencies for mne-bids, but then returns an Error. I am just going to copy the final part of the error message, please let me know if you want me to post the whole thing:

`Finished processing dependencies for mne-bids==0.9.dev0
check-manifest .
lists of files in version control and sdist match
rm -f .coverage
process_begin: CreateProcess(NULL, rm -f .coverage, ...) failed.
make (e=2): Das System kann die angegebene Datei nicht finden.
make: *** [Makefile:33: test] Error 2`

"Das System kann die angebene Datei nicht finden" translates to "The system can't found the specified file".

Reply:
you can:

1. navigate to the root of the repo, and then report to me the following outputs:

1. `git branch`
2. `git remote -v`
3. `git status`

Looking at your repository I see that you are one commit behind. This is how you get up to date:

1. make sure you have no uncommitted changes (`git status`)
1. `git checkout main`
2. `git pull`
3. `git checkout dev` (because you called your branch `dev`)
4. `git merge main`
5. Looking at your changes in your `dev` there should not be any conflicts, so this should just work

Then your branch `dev` will be up to date.

Please try `make test` again right afterwards and tell me if anything changes.

Regarding the error you report, I don't know what's wrong. I also don't see why you get a deprecation warning 🤔 

Can you try:

- `make pep`
- `pytest`
- `bids-validator --version`

And report to me what happens?

Reply:
I will quote my output below your instructions:
> 2. `git branch`
```
* dev
  main
```
> 3. `git remote -v`
```
origin  https://github.com/fravona2211/mne-bids (fetch)
origin  https://github.com/fravona2211/mne-bids (push)
```
> 4. `git status`
```
On branch dev
Your branch is up to date with 'origin/dev'.

nothing to commit, working tree clean
```
> Looking at your repository I see that you are one commit behind. This is how you get up to date:
> 
> 1. make sure you have no uncommitted changes (`git status`)
> 2. `git checkout main`
> 3. `git pull`
> 4. `git checkout dev` (because you called your branch `dev`)
> 5. `git merge main`
> 6. Looking at your changes in your `dev` there should not be any conflicts, so this should just work

`git merge main` gives me the following output: `Already up to date.`. So, am I actually a commit behind? 


> Then your branch `dev` will be up to date.
> 
> Please try `make test` again right afterwards and tell me if anything changes.

Unfortunately, no. I still get the following output: 
`process_begin: CreateProcess(NULL, rm -f .coverage, ...) failed.
make (e=2): Das System kann die angegebene Datei nicht finden.
make: *** [Makefile:33: test] Error 2`

> Regarding the error you report, I don't know what's wrong. I also don't see why you get a deprecation warning 🤔
> 
> Can you try:
> 
> * `make pep`

Here I am also running into an error: 
`make[1]: Leaving directory 'C:/Users/FvA/mne-python/mne-bids'
make: *** [Makefile:67: pep] Error 2`

> * `pytest`

Still returns the `DeprecationWarning` .

> * `bids-validator --version`

`1.8.0`

Just wondering if there might be a problem with me installing mne-bids inside of the mne-python directory? Should I reinstall it into its own?

Reply:
> Just wondering if there might be a problem with me installing mne-bids inside of the mne-python directory? Should I reinstall it into its own?

Hehe, yes, that is actually crucial. Not only from a Python perspective, but also from a Git perspective: These are two different Python packages / Git repositories. I have mine in the same subfolder, but not within each other:

```
my-subfolder/
├── mne-bids
└── mne-python
```

The remaining outputs look good, except that you are definitely one commit behind `main`. Take a look at the banner on this page: https://github.com/fravona2211/mne-bids/tree/dev and the following screenshot:

![image](https://user-images.githubusercontent.com/9084751/129519824-4935b894-7e69-4d90-9d6b-6f2810fe7190.png)


You can see something similar if you look at the `main` branch of your fork, see screenshot:

![image](https://user-images.githubusercontent.com/9084751/129519874-70c14a93-2cf2-4240-8a4e-5c9b0d493b97.png)


Your `git remote -v` command tells you what your local repository knows about other repos out there. Currently it just knows about your fork ... however as we've seen, your `main` branch of your fork is **behind** the `main` branch of mne-bids. So instead of deleting everything and making a new fork+clone, we simply update as follows:

- `git remote add upstream https://github.com/mne-tools/mne-bids`
- run again: `git remote -v`, it should now list an additional entry (a total of 4 lines, 2 per "entry")
- now `git checkout main`
- then `git pull upstream main`
- and finally `git push`

This will pull the recent changes from mne-bids "upstream", place them into your local clone of your fork, and then push them up to your repo.

Afterwards, do:

- `git checkout dev`
- `git merge main`
- `git push`

and that should work.

So there are 2 fixes that you should both try:

1. cloning mne-bids NOT within mne-python (but somewhere next to it, exactly how much next to it doesn't matter: but never clone a git repository into another git repository unless you know what you are doing)
2. updating your fork

Reply:
So I moved mne-bids next to mne-python and updated my fork. I assume this means it worked?

![image](https://user-images.githubusercontent.com/59033598/129610921-ee9cc5e1-f663-4786-bb15-4af251f5ad8d.png)

Sadly, running `pytest` still returns an error. I copied the whole error message, maybe it's easier to figure out what went wrong this way: 

```
(mne-bids) PS C:\Users\FvA\Python MNE\mne-bids> pytest
Traceback (most recent call last):
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\Scripts\pytest.exe\__main__.py", line 7, in <module>
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\config\__init__.py", line 185, in console_main
    code = main()
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\config\__init__.py", line 143, in main
    config = _prepareconfig(args, plugins)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\config\__init__.py", line 318, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\callers.py", line 203, in _multicall
    gen.send(outcome)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\helpconfig.py", line 100, in pytest_cmdline_parse
    config: Config = outcome.get_result()
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\config\__init__.py", line 1003, in pytest_cmdline_parse
    self.parse(args)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\config\__init__.py", line 1283, in parse
    self._preparse(args, addopts=addopts)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\config\__init__.py", line 1191, in _preparse
    self.hook.pytest_load_initial_conftests(
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\callers.py", line 208, in _multicall
    return outcome.get_result()
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pluggy\callers.py", line 182, in _multicall
    next(gen)  # first yield
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\capture.py", line 157, in pytest_load_initial_conftests
    _readline_workaround()
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\_pytest\capture.py", line 90, in _readline_workaround
    import readline  # noqa: F401
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\readline.py", line 34, in <module>
    rl = Readline()
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pyreadline\rlmain.py", line 422, in __init__
    BaseReadline.__init__(self)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pyreadline\rlmain.py", line 62, in __init__
    mode.init_editing_mode(None)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pyreadline\modes\emacs.py", line 633, in init_editing_mode
    self._bind_key('space',       self.self_insert)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pyreadline\modes\basemode.py", line 162, in _bind_key
    if not callable(func):
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\site-packages\pyreadline\py3k_compat.py", line 8, in callable
    return isinstance(x, collections.Callable)
  File "C:\Users\FvA\anaconda3\envs\mne-bids\lib\collections\__init__.py", line 62, in __getattr__
    warnings.warn("Using or importing the ABCs from 'collections' instead "
DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
```


Thanks so much for your extensive help!!

Reply:
yes, that worked - great. And you can remember that workflow, because it's universally used :-)

That error you get is super weird.

- what does `conda env list` give you?
- Could you maybe try a clean conda install?
- remove old: (make sure you are in the `base` env first) --> `conda env remove -n mne-bids`
- make new: `conda create -n mne_bids Python=3.8`
- `conda activate mne_bids`
- `pip install -e .` (from root of mne-bids repo)
- `pip install -r test_requirements.txt` (from root of mne-bids repo)
- `pip install -r doc/requirements.txt` (from root of mne-bids repo)
- then `cd` to your mne-python repository and call `pip install -e .` (from root of mne-python repo)

Then again, try (all from root of mne-bids repo):
- `pytest`
- `make pep`
- `make test`

Reply:
I think it's finally working, thank you so much for your help! I will try to start a new PR once I passed all the tests. 
One more question though, what's the difference between running `pytest` and `make test`? And does `make test` automatically check for style, too?

Reply:
awesome, that makes me happy @fravona2211 :-) why would you start a new PR though? you can just keep working in this one, right?

> what's the difference between running pytest and make test? And does make test automatically check for style, too?

`make test` uses GNU make and tells it to look up the `Makefile` and then perform the `test` "recipe" that is in there. You can see that recipe here:

https://github.com/mne-tools/mne-bids/blob/2c5558d0529599632e62b3067e2759fdf1f48c2e/Makefile#L29-L39

basically it's the same as calling `pytest` with a couple of extra options.

To check style, you need to call `make pep`, which will use the "pep" recipe that you can also look up in the `Makefile`.

you can also try building the docs through: `make build-doc` :-)